pkg/backup/snapshotter/snapshotter.go

```diff
@@ -486,9 +487,59 @@ func (m *StoresMixture) ProcessCSBPVCsAndPVs(r *v1alpha1.Restore, csb *CloudSnap
 		resetMetadataAndStatus(r, backupClusterName, pvc, pv)
 	}
 
+	restoreSTSName := controller.TiKVMemberName(r.Spec.BR.Cluster)
+	sequentialPVCs, err := resetPVCSequence(restoreSTSName, pvcs)
+	if err != nil {
+		klog.Errorf("reset pvcs to sequential error: %s", err.Error())
+		return "InvalidPVCName", err
+	}
+
+	csb.Kubernetes.PVCs = sequentialPVCs
 	return "", nil
 }
 
+// If a backup cluster has scaled in and the tidb-operator enabled advanced-statefulset(ref: https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/advanced-statefulset)
+// the name of pvc can be not sequential, eg: [pvc-0, pvc-1, pvc-2, pvc-6, pvc-7, pvc-8]
+// When create cluster, we should ensure pvc name sequential, so we should reset it to [pvc-0, pvc-1, pvc-2, pvc-3, pvc-4, pvc-5]
+func resetPVCSequence(stsName string, backupPVCs []*corev1.PersistentVolumeClaim) ([]*corev1.PersistentVolumeClaim, error) {
+	type indexedPVC struct {
+		index int
+		pvc   *corev1.PersistentVolumeClaim
+	}
+	indexedPVCs := make([]*indexedPVC, 0, len(backupPVCs))
+	reStr := fmt.Sprintf(`%s-(\d+)$`, stsName)
+	re := regexp.MustCompile(reStr)
+	for _, pvc := range backupPVCs {
+		subMatches := re.FindStringSubmatch(pvc.Name)
+		if len(subMatches) != 2 {
+			return nil, fmt.Errorf("pvc name %s doesn't match regex %s", pvc.Name, reStr)
+		}
+		index, err := strconv.Atoi(subMatches[1])
```

Comment:
add some comments here to help understand the code
e.g.            // pvc name "tikv-basic-tikv-1",  here we try to get the tikv instance number "1"

Reply:
done